### PR TITLE
[Material] Text scale and wide label fixes for Slider and Range Slider value indicator shape

### PR DESCRIPTION
**Pull request by clocksmith**
## Description

Various bug fixes for Slider and Range Slider value indicator around accessibility and wide labels. This change required making the neck of the value indicator dependent on a static width. This allows text scale to properly work when the value indicator paddle grows for the text, but the thumb remains the same size. The value indicator shape also gets creates an overridable method: getHorizontalOffset which calls an internal method to get the ideal offset. This is necessary so that the Range slider paint method knows whether or not the value indicators are overlapping in all cases.

Here are the cases where the current version is broken:

regular_overlap
large_overlap_middle
large_overlap_start

And here is the complete motion for the fixed version:

regular text scale:
regular

huge text scale:
huge

## Related Issues

https://github.com/flutter/flutter/issues/34734

## Tests

I added the following tests:

Added a test for top value indicator outline being painted when the text scale is double.

## Checklist

Before you create this PR confirm that it meets all requirements listed below by checking the relevant checkboxes (`[x]`). This will ensure a smooth and quick review process.

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I signed the [CLA].
- [x] I read and followed the [Flutter Style Guide], including [Features we expect every widget to implement].
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] All existing and new tests are passing.
- [x] The analyzer (`flutter analyze --flutter-repo`) does not report any problems on my PR.
- [x] I am willing to follow-up on review comments in a timely manner.

## Breaking Change

Does your PR require Flutter developers to manually update their apps to accommodate your change?

- [ ] Yes, this is a breaking change (Please read [Handling breaking changes]). *Replace this with a link to the e-mail where you asked for input on this proposed change.*
- [x] No, this is *not* a breaking change.

[issue database]: https://github.com/flutter/flutter/issues
[Contributor Guide]: https://github.com/flutter/flutter/wiki/Tree-hygiene#overview
[Test Coverage]: https://github.com/flutter/flutter/wiki/Test-coverage-for-package%3Aflutter
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[Features we expect every widget to implement]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo#features-we-expect-every-widget-to-implement
[CLA]: https://cla.developers.google.com/
[Handling breaking changes]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes

### CODE REVIEW

File: examples/flutter_gallery/lib/demo/material/slider_demo.dart [hunk 1]
```diff
@@ -438,4 +438,3 @@ class _RangeSlidersState extends State<_RangeSliders> {
     );
   }
 }
-
```
Comment (rami-a) [inline examples/flutter_gallery/lib/demo/material/slider_demo.dart]: Revert this change

File: packages/flutter/test/material/range_slider_test.dart [hunk 2]
```diff
@@ -1459,5 +1459,79 @@ void main() {
...
+    expect(values.end, closeTo(0.5, 0.03));
+    final TestGesture gesture = await tester.startGesture(middle);
+    await tester.pumpAndSettle();
+
+    expect(
+        sliderBox,
+        paints
+          ..path(color: sliderTheme.valueIndicatorColor)
+          ..path(color: sliderTheme.overlappingShapeStrokeColor)
+          ..path(color: sliderTheme.valueIndicatorColor)
```
Comment (rami-a) [inline packages/flutter/test/material/range_slider_test.dart]: Trailing comma should format this a little better

**Review by rami-a — APPROVED** [review]
LGTM

File: packages/flutter/lib/src/material/slider_theme.dart [hunk 3]
```diff
@@ -1236,6 +1236,16 @@ abstract class RangeSliderValueIndicatorShape {
   /// width because it is derived from a formatted string.
   Size getPreferredSize(bool isEnabled, bool isDiscrete, { TextPainter labelPainter });
 
+  /// Determines the best offset to keep this shape on the screen.
```
Comment (willlarche) [inline packages/flutter/lib/src/material/slider_theme.dart]: If this is intended to be overridden, maybe say so in the comment.

**Review by willlarche — APPROVED** [review]

File: packages/flutter/test/material/range_slider_test.dart (hunk 2 above)
Comment (clocksmith) [reply packages/flutter/test/material/range_slider_test.dart]: Done

File: packages/flutter/lib/src/material/slider_theme.dart (hunk 3 above)
Comment (clocksmith) [reply packages/flutter/lib/src/material/slider_theme.dart]: Done

File: packages/flutter/lib/src/material/slider_theme.dart [hunk 4]
```diff
@@ -2684,13 +2655,25 @@ class _PaddleSliderTrackShapePathPainter {
...
-      shift = parentBox.size.width - bottomRight.dx - edgeMargin;
+
+    final double endGlobal = parentBox.localToGlobal(Offset(parentBox.size.width, parentBox.size.height)).dx;
+    if (bottomRight.dx > endGlobal - edgeMargin) {
+      shift = endGlobal - edgeMargin - bottomRight.dx;
     }
+
     shift = scale == 0.0 ? 0.0 : shift / scale;
+    if (shift < 0.0) {
+      // shifting to the left
```
Comment (justinmc) [inline packages/flutter/lib/src/material/slider_theme.dart]: Capital letter and period here and below :apple: :pencil2: :100:

**Review by justinmc — APPROVED** [review]
LGTM if all the fixes here are covered by the test.

File: packages/flutter/lib/src/material/slider_theme.dart (hunk 4 above)
Comment (clocksmith) [reply packages/flutter/lib/src/material/slider_theme.dart]: Done
